data/src/history.rs

```diff
@@ -201,15 +208,18 @@ pub async fn append(
     kind: &Kind,
     messages: Vec<Message>,
     read_marker: Option<ReadMarker>,
+    reactions: HashMap<message::Id, Vec<Reaction>>,
 ) -> Result<(), Error> {
     let loaded = load(kind.clone()).await?;
 
     let mut all_messages = loaded.messages;
     messages.into_iter().for_each(|message| {
         insert_message(&mut all_messages, message);
     });
+    let mut all_reactions = loaded.metadata.reactions.into_owned();
+    all_reactions.extend(reactions);
 
-    overwrite(kind, &all_messages, read_marker).await
+    overwrite(kind, &all_messages, read_marker, &all_reactions).await
 }
```

Comment:
Isn't the full reaction history always held in memory? `append` is for appending new messages when only partial message history is open, but we should be able to just pass through `reactions` directly to `overwrite` since it should be everything.

Extending off the on-disk reactions is actually dangerous because it will re-inject things that were unreacted that we've removed from our in-memory state, but still exists on-disk. (Assuming we never store un-reacts which based on my other comment I think is the best way to go about it)